Marlin/SanityCheck.h

```diff
@@ -1155,7 +1155,7 @@ static_assert(1 >= 0
   #if ENABLED(REPRAP_DISCOUNT_FULL_GRAPHIC_SMART_CONTROLLER) && DISABLED(BQ_LCD_SMART_CONTROLLER)
     + 1
   #endif
-  #if ENABLED(LCD_FOR_MELZI)
+  #if ENABLED(LCD_FOR_MELZI) && DISABLED(REPRAP_DISCOUNT_SMART_CONTROLLER)
```

Comment:
This "fix" in conditionals is odd: if LFM and RRDSC are both enabled, isn't the warning to only enable a single LCD intentionally the correct place to catch this ?  If there's more to it, can you explain what the fix does in the PR (it's always good to do that so in case things break, it is clearer what was done so it may be clearer to fix if necessary).

Reply:
@fiveangle is correct that this isn't quite right. Whenever `LCD_FOR_MELZI` is set, so is `REPRAP_DISCOUNT_SMART_CONTROLLER`. This simply sabotages the test so it counts zero LCD options. The correct solution is actually this:
```diff
- #if ENABLED(REPRAP_DISCOUNT_SMART_CONTROLLER) && DISABLED(REPRAP_DISCOUNT_FULL_GRAPHIC_SMART_CONTROLLER)
+ #if ENABLED(REPRAP_DISCOUNT_SMART_CONTROLLER) && DISABLED(REPRAP_DISCOUNT_FULL_GRAPHIC_SMART_CONTROLLER) && DISABLED(LCD_FOR_MELZI)
```
I will apply this patch to both 1.1.x and 2.0.x shortly…

Reply:
I'll admit this is my first time digging into the Marlin source and I was a bit overwhelmed by the layers of preprocessor conditionals. Thanks for checking it out and applying the correct fix.